cryptonight/cuda_cryptonight_extra.cu

```diff
@@ -99,6 +100,15 @@ __global__ void cryptonight_extra_gpu_prepare(int threads, const uint32_t * __re
 		XOR_BLOCKS_DST(ctx_state, ctx_state + 4, ctx_a);
 		XOR_BLOCKS_DST(ctx_state + 2, ctx_state + 6, ctx_b);
 
+                if (variant > 0)
+		{
+			tweak1_2[0] = (input[8] >> 24) | (input[9] << 8);
+			tweak1_2[0] ^= (ctx_state[24] & 0xffffffff);
```

Comment:
Why do you need to to use ` & 0xffffffff`, `ctx_state` is already a 32bit value.

Reply:
Thats my fault I looked to the wrong definition `ctx_state` is of the type `uin64_t`.